libnavigation-core/src/test/java/com/mapbox/navigation/core/MapboxNavigationTest.kt

```diff
@@ -1391,89 +1188,18 @@ class MapboxNavigationTest {
             }
         }
 
-    private fun createMapboxNavigation() {
-        mapboxNavigation = MapboxNavigation(navigationOptions, threadController)
-    }
-
-    private fun mockNativeNavigator() {
-        every {
-            NavigationComponentProvider.createNativeNavigator(
-                any(),
-                any(),
-                any(),
-                any(),
-                any(),
-                any(),
-            )
-        } returns navigator
-        coEvery { navigator.setRoutes(any(), any(), any()) } answers {
-            createSetRouteResult()
-        }
-    }
-
-    private fun mockTripService() {
-        every {
-            NavigationComponentProvider.createTripService(
-                applicationContext,
-                any(),
-                threadController,
-            )
-        } returns tripService
-    }
-
-    private fun mockTripSession() {
-        every {
-            NavigationComponentProvider.createTripSessionLocationEngine(
-                navigationOptions = navigationOptions
-            )
-        } returns tripSessionLocationEngine
-
-        every {
-            NavigationComponentProvider.createTripSession(
-                tripService = tripService,
-                tripSessionLocationEngine = tripSessionLocationEngine,
-                navigator = navigator,
-                threadController,
-            )
-        } returns tripSession
-        every { tripSession.getRouteProgress() } returns routeProgress
-        coEvery { tripSession.setRoutes(any(), any(), any()) } returns NativeSetRouteResult(
-            nativeAlternatives = emptyList()
-        )
-    }
-
-    private fun mockDirectionSession() {
-        every { NavigationComponentProvider.createDirectionsSession(any()) } answers {
-            directionsSession
-        }
-        // TODO Needed for telemetry - Free Drive (empty list) for now
-        every { directionsSession.routes } returns emptyList()
+    @Test
+    fun `set reroute controller`() = coroutineRule.runBlockingTest {
+        createMapboxNavigation()
+        mapboxNavigation.setRerouteController(rerouteController)
+        assertTrue(mapboxNavigation.getRerouteController() is LegacyRerouteControllerAdapter)
```

Comment:
why do you need to check this implementation detail? This test will fail during refactoring even if the feature works.